src/string_scanner.cr

```diff
@@ -280,6 +282,34 @@ class StringScanner
     @str.byte_slice(@byte_offset, @str.bytesize - @byte_offset)
   end
 
+  # Returns one byte from the current offset.
+  # ```
+  # require "string_scanner"
+  #
+  # s = StringScanner.new("あ")
+  # s.read_byte # => "\xE3"
+  # s.read_byte # => "\x81"
+  # s.read_byte # => "\x82"
+  # ```
+  def read_byte : String?
+    return nil if eos?
+    s = @str.byte_slice(@byte_offset, 1)
+    @byte_offset += 1
+    s
+  end
+
+  # Returns one char from the current offset.
+  # ```
+  # require "string_scanner"
+  #
+  # s = StringScanner.new("ab")
+  # s.read_char # => "a"
+  # s.read_char # => "b"
+  # ```
+  def read_char : String?
+    scan(/./)
```

Comment:
`/./` might actually miss a few characters. The safest way is to do it in Crystal with something like `String#char_bytesize_at`.

Reply:
How about `/./m`? `/./` misses newline characters, so changing it multiline mode, it detects newline characters. It also changes behaviors of `^` and `$`, but it doesn't matter.